sagemaker-python-sdk/tensorflow_serving_using_elastic_inference_with_your_own_model/tensorflow_neo_compiled_model_elastic_inference.ipynb

```diff
@@ -0,0 +1,227 @@
+    "    for file in os.listdir(saved_model_dir):\n",
+    "        tar.add(os.path.join(saved_model_dir, file), arcname=file)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Upload model artifact to S3 bucket\n",
+    "Amazon SageMaker Neo expects a path to the model artifact in Amazon S3, so we will upload the model artifact to be compiled to S3 bucket."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from sagemaker.utils import name_from_base\n",
+    "\n",
+    "compilation_job_name = name_from_base('Keras-ResNet50')\n",
+    "input_model_path = session.upload_data(path='model.tar.gz', bucket=bucket, key_prefix=compilation_job_name)\n",
+    "print('S3 path for input model: {}'.format(input_model_path))"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Compile model for EI accelerator using Neo\n",
+    "Now the model is ready to be compiled by Neo. Note that `ml_eia2` needs to be set for `target_isntance_family` field in order for the model to be optimized for EI accelerator. If you want to compile your own model for EI accelerataor, refer to [Neo compilation API](https://sagemaker.readthedocs.io/en/stable/api/inference/model.html#sagemaker.model.Model.compile) to provide the proper `input_shape` and optional `compiler_options` according to your model.\n",
+    "\n",
+    "**Important: If the following command result in a permission error, scroll up and locate the value of execution role returned by get_execution_role(). The role must have access to the S3 bucket specified in output_path.**"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from sagemaker.tensorflow import TensorFlowModel\n",
+    "\n",
+    "# Create a TensorFlow SageMaker model\n",
+    "tensorflow_model = TensorFlowModel(model_data=input_model_path,\n",
+    "                                   role=role,\n",
+    "                                   framework_version='2.3')\n",
+    "\n",
+    "# Compile the model for EI accelerator in SageMaker Neo\n",
+    "output_path = '/'.join(input_model_path.split('/')[:-1])\n",
+    "tensorflow_model.compile(target_instance_family='ml_eia2',\n",
+    "                         input_shape={\"input_1\": [1, 224, 224, 3]},\n",
+    "                         output_path=output_path,\n",
+    "                         role=role,\n",
+    "                         job_name=compilation_job_name,\n",
+    "                         framework='tensorflow')"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Deploy compiled model to SageMaker Endpoint with EI accelerator attached\n",
+    "There is no difference between deploying Neo-compiled model and uncompiled model to EI-attached SageMaker endpoint.\n",
```

Comment:
Possible re-wording:
"The same methods are used to deploy a model to a SageMaker Endpoint with EI regardless of whether or not the model is compiled or not compiled by Neo.'